x-pack/plugins/cases/public/components/all_cases/columns.test.tsx

```diff
@@ -10,51 +10,67 @@ import { mount } from 'enzyme';
 
 import '../../common/mock/match_media';
 import { ExternalServiceColumn } from './columns';
-
 import { useGetCasesMockState } from '../../containers/mock';
+import { useKibana } from '../../common/lib/kibana';
+import { actionTypeRegistryMock } from '../../../../triggers_actions_ui/public/application/action_type_registry.mock';
+import { connectors } from '../configure_cases/__mock__';
 
-jest.mock('../../common/lib/kibana', () => {
-  const originalModule = jest.requireActual('../../common/lib/kibana');
-  return {
-    ...originalModule,
-    useKibana: () => ({
-      services: {
-        triggersActionsUi: {
-          actionTypeRegistry: {
-            get: jest.fn().mockReturnValue({
-              actionTypeTitle: '.jira',
-              iconClass: 'logoSecurity',
-            }),
-          },
-        },
-      },
-    }),
-  };
-});
+jest.mock('../../common/lib/kibana');
+const useKibanaMock = useKibana as jest.Mocked<typeof useKibana>;
 
 describe('ExternalServiceColumn ', () => {
+  const { createMockActionTypeModel } = actionTypeRegistryMock;
+
+  beforeAll(() => {
+    connectors.forEach((connector) =>
+      useKibanaMock().services.triggersActionsUi.actionTypeRegistry.register(
+        createMockActionTypeModel({ id: connector.actionTypeId, iconClass: 'logoSecurity' })
+      )
+    );
+  });
+
   it('Not pushed render', () => {
     const wrapper = mount(
-      <ExternalServiceColumn {...{ theCase: useGetCasesMockState.data.cases[0] }} />
+      <ExternalServiceColumn theCase={useGetCasesMockState.data.cases[0]} connectors={connectors} />
     );
     expect(
       wrapper.find(`[data-test-subj="case-table-column-external-notPushed"]`).last().exists()
     ).toBeTruthy();
   });
+
   it('Up to date', () => {
     const wrapper = mount(
-      <ExternalServiceColumn {...{ theCase: useGetCasesMockState.data.cases[1] }} />
+      <ExternalServiceColumn theCase={useGetCasesMockState.data.cases[1]} connectors={connectors} />
     );
     expect(
       wrapper.find(`[data-test-subj="case-table-column-external-upToDate"]`).last().exists()
     ).toBeTruthy();
   });
+
   it('Needs update', () => {
     const wrapper = mount(
-      <ExternalServiceColumn {...{ theCase: useGetCasesMockState.data.cases[2] }} />
+      <ExternalServiceColumn theCase={useGetCasesMockState.data.cases[2]} connectors={connectors} />
     );
     expect(
       wrapper.find(`[data-test-subj="case-table-column-external-requiresUpdate"]`).last().exists()
     ).toBeTruthy();
   });
+
+  it('it does not throw when accessing the icon if the connector type is not registered', () => {
```

Comment:
Just to double check, does the mock implementation of the trigger actions service still throw when it can't find the connector?

Reply:
Actually, they do not mock the registry. So yes, it will throw if the connector is not there. You can try that by removing the registering of the connectors at the top of the file.